### PR TITLE
Introduced Error Message to Task Execution

### PR DESCRIPTION
**Pull request by mminella**
To provide the ability to control task orchestration via the exit
message, we moved the stack trace storage to a new field.  This means
that exit message is now a field we can use similar to how exit code is
in batch, and still have the stack trace of an unsuccessful task
execution stored.

Resolves spring-cloud/spring-cloud-task#186

### CODE REVIEW

File: spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/TaskExecutionDao.java [hunk 1]
```diff
@@ -49,6 +49,17 @@ TaskExecution createTaskExecution( String taskName,
 	 * @param exitCode the status of the task upon completion.
 	 * @param endTime the time the task completed.
 	 * @param exitMessage the message assigned to the task upon completion.
```
Comment (cppwfs) [inline spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/TaskExecutionDao.java]: Need to add errorMessage to javadoc.   Will fix on merge.

File: spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskExecutionListenerTests.java [hunk 2]
```diff
@@ -155,7 +155,7 @@ public void testAnnotationFail() {
 		context.publishEvent(new ApplicationReadyEvent(application, new String[0], context));
 
 		TaskExecution taskExecution = new TaskExecution(0, 1, "wombat", new Date(),
-				new Date(), null, new ArrayList<String>());
+				new Date(), null, new ArrayList<String>(), null);
```
Comment (cppwfs) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskExecutionListenerTests.java]: Probably want to check the errorMessage in the verifyListenerResults.  I can do this on merge.

File: spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java [hunk 3]
```diff
@@ -139,7 +139,7 @@ private void verifyTaskExecution(int numberOfParams, boolean update, Integer exi
 		assertEquals(exitCode, taskExecution.getExitCode());
 
 		if(exception != null) {
-			assertTrue(taskExecution.getExitMessage().length() > exception.getStackTrace().length);
+			assertTrue(taskExecution.getErrorMessage().length() > exception.getStackTrace().length);
```
Comment (cppwfs) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java]: probably want to have the assertNull on the errorMessage on line 145.  its checking exit message.  will fix on merge.

File: spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java [hunk 4]
```diff
@@ -104,6 +107,8 @@ public void setup() {
 		properties.put("spring.datasource.password", DATASOURCE_USER_PASSWORD);
 		properties.put("spring.datasource.driverClassName", DATASOURCE_DRIVER_CLASS_NAME);
 		properties.put("spring.application.name",TASK_NAME);
+		properties.put("spring.cloud.task.initialize.enable", "false");
+//		properties.put("logging.level.org.springframework.cloud.task", "DEBUG");
```
Comment (cppwfs) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java]: remove this commented out line.   will fix on merge.

File: spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java [hunk 5]
```diff
@@ -146,6 +161,7 @@ private boolean tableExists() throws SQLException {
 	private boolean waitForDBToBePopulated() throws Exception {
 		boolean isDbPopulated = false;
 		for (int waitTime = 0; waitTime <= MAX_WAIT_TIME; waitTime += WAIT_INTERVAL) {
+			System.out.println(">> Table exists: " + tableExists() + " execution count = " + taskExplorer.getTaskExecutionCount());
```
Comment (cppwfs) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java]: remove the system.out.   Will fix on merge.

File: spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/TaskExecutionDao.java (hunk 1 above)
Comment (mminella) [inline spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/TaskExecutionDao.java]: Agreed

File: spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskExecutionListenerTests.java (hunk 2 above)
Comment (mminella) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskExecutionListenerTests.java]: Agreed

File: spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java (hunk 3 above)
Comment (mminella) [inline spring-cloud-task-core/src/test/java/org/springframework/cloud/task/listener/TaskLifecycleListenerTests.java]: Agreed

File: spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java (hunk 4 above)
Comment (mminella) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java]: Agreed

File: spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java (hunk 5 above)
Comment (mminella) [inline spring-cloud-task-integration-tests/src/test/java/org/springframework/cloud/task/launcher/TaskLauncherSinkTests.java]: Agreed

**Comment by cppwfs** [comment]
LGTM.  Cleaned up requested changes above.  Squashed, Rebased, Merged.

**Pull request closed without merging**